modules/apps/29-fee/keeper/relay.go

```diff
@@ -19,11 +20,17 @@ func (k Keeper) SendPacket(ctx sdk.Context, chanCap *capabilitytypes.Capability,
 func (k Keeper) WriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI, acknowledgement []byte) error {
 	// retrieve the forward relayer that was stored in `onRecvPacket`
 	packetId := channeltypes.NewPacketId(packet.GetSourceChannel(), packet.GetSourcePort(), packet.GetSequence())
+
+	// relayer address returned here is the
 	relayer, _ := k.GetForwardRelayerAddress(ctx, packetId)
+	forwardRelayer, found := k.GetCounterpartyAddress(ctx, relayer)
```

Comment:
We want to ignore the error here.

This is because its possible that relayer has not registered counterparty address. If they haven't then, we just write acknowledgement with an empty relayer address.

We don't want to prevent writeacknowledgement from succeeding because relayer hasn't registered counterparty address.

This effectively means that recv_fee will get refunded and not incentivized, but that's the relayer's fault

Reply:
Done